components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+        }
+        String result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getDeviceOrganizationParent(deviceId);
+            if (result == null) {
+                String msg = "Device with ID: " + deviceId + " not found.";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving state of device with ID:" + deviceId + " to Device Organization ";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/{deviceId}/isgateway")
+    @Override
+    public Response getDeviceOrganizationIsGateway(@PathParam("deviceId") String deviceId) {
+        if (deviceId == null) {
+            String errorMessage = "The parameter of the device organization ID is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        int result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getDeviceOrganizationIsGateway(deviceId);
+            if (result == -1) {
+                String msg = "Device with ID: " + deviceId + " not found.";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving id device with ID:" + deviceId + " is gateway in Device Organization ";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/devices")
+    @Override
+    public Response getDevicesInOrganization() {
+        List<DeviceOrganizationMetadataHolder> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getDevicesInOrganization();
+            if (result == null) {
+                String msg = "No devices exist in organization";
+                log.error(msg);
```

Comment:
IMO, this is not an error. We can log it as a debug log.

Reply:
Fixed ce1b79e04ff7cb44e6a64e88ebab322946c55f47